Added data sending to the backend of the create page

ahmetackk opened the pull request:
Data is not lost between tabs and the save button can give a warning to enter missing data.

example of data that sending to the backend

{
  "basics": {
    "category": {
      "primaryCategory": "Technology",
      "primarySubcategory": "AI",
      "secondaryCategory": "Games",
      "secondarySubcategory": "Video Games"
    },
    "title": "Image to Video AI",
    "subtitle": "Turn your images to video",
    "location": "Gebze, Kocaeli, Türkiye",
    "image": [
      "blob:http://localhost:5173/0df383b0-7ec1-4f5f-9024-acd3e14a820b",
      "blob:http://localhost:5173/32926565-be8f-4685-8be5-f8a098883471",
      "blob:http://localhost:5173/6676b25b-fd3f-49a7-bafe-b2281bc5d9e6"
    ],
    "video": "blob:http://localhost:5173/b6025a15-b130-47f9-81dc-9383620463ca",
    "targetDate": "2024-12-29T21:00:00.000Z",
    "duration": {
      "type": "fixed",
      "value": "50"
    }
  },
  "funding": {
    "currency": "USD",
    "amount": "2000"
  },
  "story": {
    "story": [
      {
        "id": 0,
        "title": "Give us images and watch them turn to a video",
        "content": "With ai we can create videos from your images"
      }
    ]
  },
  "collaborators": {
    "collaborators": "{ahmetaci@email.com, Ahmet Acıkök, {Edit project, Manage community}}"
  }
}

ahmetackk committed:
Added data sending to the backend of the create page

Data is not lost between tabs and the save button can give a warning to enter missing data.

umutsatir commented:
Stories must be like this:

{
  "story": [
    {
      "heading": "Introduction",
      "paragraphs": [
        "In the modern era, software projects have become a cornerstone of innovation and problem-solving.",
        "This story takes you through the lifecycle of a software project, from inception to deployment."
      ]
    },
    {
      "heading": "Planning Phase",
      "paragraphs": [
        "Every great project begins with an idea, followed by thorough planning to turn that idea into reality.",
        "Stakeholders brainstorm, set objectives, and define the scope of work during this phase."
      ]
    },
    {
      "heading": "Development",
      "paragraphs": [
        "The development phase is where the magic happens. Developers write code, integrate systems, and build features.",
        "Challenges are inevitable, but collaboration and problem-solving are key to overcoming them."
      ]
    },
    {
      "heading": "Testing and Feedback",
      "paragraphs": [
        "Testing ensures the project meets quality standards and user expectations.",
        "Feedback loops during this phase help fine-tune the project for a successful launch."
      ]
    },
    {
      "heading": "Deployment and Maintenance",
      "paragraphs": [
        "Once the project is ready, it's deployed to production for users to access.",
        "Continuous maintenance and updates ensure the software remains relevant and functional."
      ]
    }
  ]
}

umutsatir reviewed (changes requested):
please read my previous comment

ahmetackk committed:
Story component updated

Added multiple content feature to story component also updated data that sending to the backend

ahmetackk commented:
updated data that sending to the backend

"story": [
      {
        "id": 0,
        "title": "tit",
        "paragraphs": [
          "adfsafsd",
          "asdfa"
        ]
      },
      {
        "id": 1,
        "title": "asd",
        "paragraphs": [
          "fdasfas"
        ]
      },
      {
        "id": 2,
        "title": "fassfadas",
        "paragraphs": [
          "fsadafd",
          "dfasfas",
          "dasfas",
          "asdf"
        ]
      }
    ]
    
    id's are using by component to make updates on the data usage of id were not changed to avoid an error in case 2 title the same
    
    
   Here is updated story item
![image](https://github.com/user-attachments/assets/fc7c6d39-8bad-451a-ab8c-ee0005d3d38b)

All you have to do is press enter to change the new paragraph field. To delete, pressing backspace while the paragraph section is empty will delete the added paragraph section.

ahmetackk committed:
Update CreateTab.jsx

ahmetackk committed:
Notification Component Added

There is 4 type for notifications: warning, success, error and info.

ahmetackk commented:
pictures of notification component:

![image](https://github.com/user-attachments/assets/4626de77-3205-4ac5-a346-0c20656aca69)



example usage of notification component:

 `import React, { useRef } from 'react';
  import Notification from "../components/Notification/Notification";  
  
  function NotifPage() {
      const notificationRef = useRef();
  
      const handleNotification = (msg, type) => {
          notificationRef.current.addNotification(msg, type);
      };
  
      return (
          <div>
              <button onClick={() => handleNotification('Success', 'success')}>
                  Success
              </button>
              <button onClick={() => handleNotification('Error!', 'error')}>
                  Error
              </button>
              <button onClick={() => handleNotification('Warning!', 'warning')}>
                  Warning
              </button>
              <button onClick={() => handleNotification('Info!', 'info')}>
                  Info
              </button>
              <Notification ref={notificationRef} />
          </div>
      );
  }
  export default NotifPage;`

umutsatir commented:
Well done! Thanks for your work and clean description!
